File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/source/local/database/AppDatabase.java
package com.ryanrvldo.moviecatalogue.data.source.local.database;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.ryanrvldo.moviecatalogue.data.source.model.Movie;
import com.ryanrvldo.moviecatalogue.data.source.model.Search;
import com.ryanrvldo.moviecatalogue.data.source.model.TvShow;


@Database(entities = {Movie.class, TvShow.class, Search.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    public abstract MovieDao movieDao();

    public abstract TvShowDao tvShowDao();

    public abstract SearchDao searchDao();

    private static volatile AppDatabase instance = null;

    @NonNull
    public static synchronized AppDatabase getInstance(Context context) {
        if (instance == null) {
            synchronized (AppDatabase.class) {
                if (instance == null) {
                    instance = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, "movie_catalogue_db")
                            .fallbackToDestructiveMigration()
                            .build();
                }
            }
        }
        return instance;
    }
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/data/local/database/AppDatabase.kt
package com.ryanrvldo.moviecatalogue.data.local.database

import androidx.room.Database
import androidx.room.RoomDatabase
import com.ryanrvldo.moviecatalogue.data.model.Movie
import com.ryanrvldo.moviecatalogue.data.model.Search
import com.ryanrvldo.moviecatalogue.data.model.TvShow

@Database(
    entities = [Movie::class, TvShow::class, Search::class],
    version = 1,
    exportSchema = false
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun movieDao(): MovieDao
    abstract fun tvShowDao(): TvShowDao
    abstract fun searchDao(): SearchDao
}